core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/GraphicElementFactory.java

```diff
@@ -187,15 +203,38 @@ public SettingBundle getFavoriteLookSettings() {
    */
   public final void setLook(String lookName) {
     // get the customer lookSettings
-    String selectedLook = lookSettings.getString(lookName, null);
-    if (selectedLook == null) {
-      // ce look n'existe plus, look par defaut
-      selectedLook = defaultLook;
+    String look = lookSettings.getString(lookName, null);
+    if (!isLookAllowed(lookName)) {
+      // this look no more exists, set an existing look
+      lookName = getAvailableLooksForUser().get(0);
+      look = lookSettings.getString(lookName, null);
     }
+    this.favoriteLookSettings = ResourceLocator.getSettingBundle(look);
+    this.currentLookName = lookName;
+  }
 
-    this.favoriteLookSettings = ResourceLocator.getSettingBundle(selectedLook);
+  private static String getExistingLook() {
+    Set<String> lookNames = lookSettings.keySet();
+    for (String aLookName : lookNames) {
+      if (isLookExist(aLookName)) {
+        return aLookName;
+      }
+    }
+    return defaultLookName;
+  }
 
-    currentLookName = lookName;
+  private static boolean isLookExist(String lookName) {
+    try {
+      String selectedLook = lookSettings.getString(lookName);
+      SettingBundle look = ResourceLocator.getSettingBundle(selectedLook);
+      return look.exists();
+    } catch (MissingResourceException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1166)